### ISSUE TITLE
(why) does the "add recycling materials" quest delete some recycling tags which SC doesn't know?

### ISSUE THREAD

**Issue by Lukas458** [westnordost/StreetComplete#2156]
Hello,

I have a question to this changeset with the recycling materials quest from SC: https://www.openstreetmap.org/changeset/92312150

Why have the previous recycling tags been deleted in this changeset?
I'm not so fine with this. One the one hand, something like _recycling:paper_packaging_ can "fall" under _recycling:paper_, yes, okay. But why do _recycling:cardboard_ or _recycling:shoes_ have been deleted, too? Only because the user didn't pick these options in the app? Maybe he forgot to pick _cardboard_, or isn't this part of the selectable features? For me, cardboard or cartons are still not the same as paper. I mean, in other case not two or three tags would have been created.

I do not understand why information has been taken away there.

**Proposed solution**
Just leave recycling tags which the app doesn't know as well as SC doesn't support them. For example, when there is decided that _recycling:paper_ means the same as _recycling:cardboard_ (where I still think it doesn't mean exactly the same), then just leave _recycling:cardboard_ there, only remove it when _recycling:paper_ was picked for "no".
Otherwise, SC will take _recycling:cardboard_ away from all containers it was asked for, and I don't know why this should be necessary.

**Comment by westnordost** [comment]
> Only because the user didn't pick these options in the app? 

Yes, this is exactly the reason. The user is supposed to select **all** that can be recycled here and if there is something he cannot select, he is supposed to leave a note. If the app were to only add but not remove recyclings that the user did not select, this would not be a very effective re-survey because only more and more would seemingly be recycled but not less.

Below the question, it is explicitly stated:

**_"If anything different can be recycled than selectable here, please leave a note instead."_**

In detail to what you wrote:

- `recycling:shoes` is supported by StreetComplete. The user did not select shoes.
- `recycling:cardboard` is a subcategory of `recycling:paper`. It is not supported because it is contained in `recycling:paper`. If he selected paper, cardboard is implicitly contained in this. If there is really a place that only accepts cardboard and not other kind of paper, it will explicitly be written on the container in which case the user should leave a note because he cannot select cardboard in the app

**Comment by westnordost** [comment]
To further elaborate on this. If there is a container, and it says "(All kinds of) Paper", then I believe it is valid and in fact even the only correct thing to do to tag `recycling:paper=yes` and only `recycling:paper=yes`. **Not** `recycling:cardboard=yes`, `recycling:cartons=yes`, `recycling:magazines=yes`, `recycling:newspaper=yes`, `recycling:paper_packaging=yes` etc.

In short, tag what is written on the container.

This has a very simple and plausible reason:

`recycling:*` is a bit of a free-form tag, there are people always adding new recycling types when they find places that accept specifically these. The way the `recycling:*` scheme is designed, this is perfectly possible. But this also means that each `recycling:*` stands for its own and you must tag what you see, *tag the type of container**, not what all can be recycled in that container.

Here is one example that illustrates this: Let's say I find a place that specifically recycles LEGOs, `recycling:lego=yes`. This is the same way `recycling:newspaper=yes` etc. came into existance - someone saw a place where he could recycle newspaper and only newspaper and he tagged and documented it.
So if the `recycling:*` scheme was to be understood that you should tag all that can be recycled at a place rather than what is written on the container, then latest when I document `recycling:lego=yes`, it would mean that I just **redefined** what needs to be tagged on a container that just accepts any kind of plastic (`recycling:plastic=yes`), so I basically invalidated [over 30000](https://taginfo.openstreetmap.org/keys/recycling%3Aplastic) recycling containers that accept plastic.
Luckily it does not mean that, otherwise the `recycling:*` tagging scheme would be highly problematic.

`recycling:*` is the same tagging scheme as [`ramp:*`](https://wiki.openstreetmap.org/wiki/Key:ramp) which by the way I also misunderstood first, thinking I have to tag `ramp:bicycle=yes` if there is a stroller ramp because I can use that stroller ramp also for bicycles. No, `ramp:*` specifies the type of ramp, not that "this and that vehicle can also use that ramp".

**TL;DR:** `recycling:paper=yes` means that there is a paper recycling container at this recycling facility, `recycling:newspaper` means that there is a newspaper recycling container, not that "newspaper can also be recycled in that paper container".

**Comment by Lukas458** [comment]
Oh, okay, I can see your point now better, yes. Thanks for your explicit explanation. Okay so I think we can close this.

**Comment by Lukas458** [comment]
Maybe someone wants to add somethng of your argumentation to the wiki. I don't know but I like how you explained the tagging scheme for recycling:=*. Maybe this can help also other users to understand that they'll have not to tag all maetrials to a container if something is covered for example by recycling:paper.

**Comment by westnordost** [comment]
I've read the recycling article and don't find anything particular to improve right now. It already mentions the "subcategories", the issue with implicit redefinition of values is more of a basic principle of OSM tagging, not sure how or if it should mentioned there.

However, I've reworked the ramp-article a bit.
